FF1Lib/asm/11_B006_GiveReward.asm

```diff
@@ -0,0 +1,66 @@
+;; Bank 11 $B006
+;; New jump point for NPC-only items:
+;; IN:
+;;       'A' should be set to the item ID, 224-255 are special values for variables
+;; Result:
+;;       'C' set if can't carry any more, otherwise clear
+;;       'X' Dialog ID
+;;       'A' Also Dialog ID
+GiveReward:                    ; (8 bytes)
+    STA dlg_itemid             ; record that as the item id so it can be printed in the dialogue box 8561
+    ADC #$20                   ; Add 0x20 to account for the 'items' offset 6920
+    CMP #$3C                   ; see if the ID is >= item_stop C93C
+    BCS @NotItem               ; B028
+  @Item:                       ; (5 + 15 + 7 bytes)
+    TAX                        ; put item ID in X AA
+    CMP #$0C                   ; then check for canal C90C
+    BNE :+                     ; If canal then D005
+      DEC unsram, X            ; decrement DE0060
+      BCS @WasCanal            ; and open it B018
+:   CMP #$36                   ; CMP ItemId with Tent C936
+    BCC :+                     ; < Tent normal Add One routine 9011
+		SBC #$36               ; Subtract TentOffset E936
+		TAY                    ; Punt Index into Y A8
```

Comment:
sometime the game expect registers to be left alone, do we need to backup Y's value here?